### PR TITLE
[tinytiff] add new port

### PR DESCRIPTION
**Pull request by EfesX**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

#38778
jkriege2/TinyTIFF#29

### CODE REVIEW

File: ports/tinytiff/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
+  "features": {
+    "winapi": {
+      "description": "Use WIN-API for File-IO"
```
Comment (dg0yt) [inline ports/tinytiff/vcpkg.json:19]:
```suggestion
      "description": "Use WIN-API for File-IO",
      "supports": "windows"
```

File: ports/tinytiff/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        "winapi"    TinyTIFF_USE_WINAPI_FOR_FILEIO
+)
+
+if(NOT VCPKG_TARGET_IS_WINDOWS AND TinyTIFF_USE_WINAPI_FOR_FILEIO)
+    message(FATAL_ERROR "Can't build ${PORT}:${TARGET_TRIPLET} with 'winapi' feature.")
+endif()
+
```
Comment (dg0yt) [inline ports/tinytiff/portfile.cmake:10]:
```suggestion
```
thanks to `supports` in manifest.

File: ports/tinytiff/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,49 @@
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        "winapi"    TinyTIFF_USE_WINAPI_FOR_FILEIO
```
Comment (dg0yt) [inline ports/tinytiff/portfile.cmake:4]:
```suggestion
        winapi      TinyTIFF_USE_WINAPI_FOR_FILEIO
```

File: ports/tinytiff/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    SHA512 28fb3d1ef1630a4d20da021ccca93f99df8bd29462525be312dfb028239176ca940a43407b2db10488d891a1fbca65d8a59bc6cc097765389f35021e8b423885
+    HEAD_REF master
+)
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    set(BUILD_SHARED_LIBS OFF)
+else()
+    set(BUILD_SHARED_LIBS ON)
+endif()
+
```
Comment (dg0yt) [inline ports/tinytiff/portfile.cmake:24]:
```suggestion
```
Handled by `vcpkg_cmake_configure`.

File: ports/tinytiff/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        ${FEATURE_OPTIONS}
+        -DTinyTIFF_BUILD_TESTS=OFF
+        -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
```
Comment (dg0yt) [inline ports/tinytiff/portfile.cmake:30]:
```suggestion
```
Handled by `vcpkg_cmake_configure`.

File: ports/tinytiff/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        -DTinyTIFF_USE_WINAPI_FOR_FILEIO=${TinyTIFF_USE_WINAPI_FOR_FILEIO}
```
Comment (dg0yt) [inline ports/tinytiff/portfile.cmake:31]:
```suggestion
```
Included in `FEATURE_OPTIONS`.

File: ports/tinytiff/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    PATCHES
+        # without this patch, the MSVC compiler will crash during the build process 
+        "msvc-message-support.patch"
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/TinyTIFF DO_NOT_DELETE_PARENT_CONFIG_PATH)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/TinyTIFFXX)
```
Comment (dg0yt) [inline ports/tinytiff/portfile.cmake:40]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/TinyTIFFXX PACKAGE_NAME tinytiffxx)
```
(guessed, and changed to lower case)

File: ports/tinytiff/msvc-message-support.patch [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+@@ -56,7 +56,11 @@
+ 
+ #ifdef TINYTIFF_USE_WINAPI_FOR_FILEIO
+ #  include <windows.h>
+-#  warning COMPILING TinyTIFFReader with WinAPI
++#  ifdef _MSC_VER
++#    pragma message(__FILE__ "(): COMPILING TinyTIFFWriter with WinAPI")
++#  else
++#    warning COMPILING TinyTIFFWriter with WinAPI
++#  endif // _MSC_VER
```
Comment (dg0yt) [inline ports/tinytiff/msvc-message-support.patch:14]: Maybe it is enough to remove the original lines? Nobody will read the logs, after asking for the feature.

Comment (EfesX) [reply ports/tinytiff/msvc-message-support.patch:14]: @dg0yt  Thank you for your review. I will make the necessary changes to the patch.

File: ports/tinytiff/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        ${FEATURE_OPTIONS}
+        -DTinyTIFF_BUILD_TESTS=OFF
+    PATCHES
+        # without this patch, the MSVC compiler will crash during the build process 
+        "msvc-support.patch"
```
Comment (FrankXie05) [inline ports/tinytiff/portfile.cmake:22]:
```suggestion
        "msvc-support.patch" # without this patch, the MSVC compiler will crash during the build process
```
If this patch is not for vcpkg, then there needs to be a report.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/tinytiff/portfile.cmake (hunk 9 above)
Comment (EfesX) [reply ports/tinytiff/portfile.cmake:22]: @FrankXie05 Thanks for the review. 

 This patch is necessary to resolve a compatibility issue with the MSVC compiler and winapi feature. 

I have also submitted a pull request (jkriege2/TinyTIFF#30), which reports about this issue.

**Comment by jkriege2** [comment]
I can make a Bugfix release!

**Comment by jkriege2** [comment]
THere you go: https://github.com/jkriege2/TinyTIFF/releases/tag/4.0.1.0

**Comment by EfesX** [comment]
@jkriege2 Thanks!

**Review by FrankXie05 — APPROVED** [review]

File: versions/t-/tinytiff.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+      "git-tree": "43a5cc041d063ffcaacb5bc8d32afe577ae760ab",
+      "version": "4.0.1.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "f88fe698653bfc0927c66837a909652e57021b40",
+      "version": "4.0.0.0",
+      "port-version": 0
+    }
+  ]
```
Comment (FrankXie05) [inline versions/t-/tinytiff.json:13]:
```suggestion
    }
```
We only need the latest version record. :)

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
The usage has been tested successfully on `x64-windows`.

File: ports/tinytiff/vcpkg.json [hunk 11]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "tinytiff",
+  "version": "4.0.1.0",
+  "description": "Lightweight TIFF reader/writer library",
+  "homepage": "http://jkriege2.github.io/TinyTIFF/",
```
Comment (sidy3d) [inline ports/tinytiff/vcpkg.json:5]: Link should be https://

**Review by vicroms — APPROVED** [review]

**Pull request merged**